packages/teams-js/test/public/settings.spec.ts

```diff
@@ -71,46 +70,136 @@ describe('settings', () => {
     });
   });
 
-  it('should successfully get settings', done => {
-    utils.initializeWithContext('settings').then(() => {
-      settings.getSettings(settings => {
-        expect(settings).toBe(expectedSettings);
-        done();
-      });
+  describe('Testing settings.getSettings function', () => {
+    const allowedContexts = [
+      FrameContexts.content,
+      FrameContexts.settings,
+      FrameContexts.remove,
+      FrameContexts.sidePanel,
+    ];
+    const expectedSettings: settings.Settings = {
+      suggestedDisplayName: 'someSuggestedDisplayName',
+      contentUrl: 'someContentUrl',
+      websiteUrl: 'someWebsiteUrl',
+      entityId: 'someEntityId',
+    };
+
+    it('settings.getSettings should not allow calls before initialization', () => {
+      expect(() => {
+        settings.getSettings(settings => {
+          expect(settings).toBe(expectedSettings);
+        });
+      }).toThrowError('The library has not yet been initialized');
+    });
 
-      const message = utils.findMessageByFunc('settings.getSettings');
-      expect(message).not.toBeNull();
+    Object.values(FrameContexts).forEach(context => {
+      if (!allowedContexts.some(allowedContexts => allowedContexts === context)) {
+        it(`settings.getSettings does not allow calls from ${context} context`, async () => {
+          await utils.initializeWithContext(context);
+          expect(() => settings.getSettings(emptyCallback)).toThrowError(
```

Comment:
Can you verify if this works as intended? Sometimes `toThrowError` provides a positive test pass when the output isn't correct.

Reply:
Yes, I have verified the test case in both positive test pass and fail.

Reply:
Awesome